packages/core/src/utils.ts

```diff
@@ -54,17 +54,28 @@ export function createReference<T extends Resource>(resource: T): Reference<T> {
  * @param resource The FHIR resource.
  * @returns A reference string of the form resourceType/id.
  */
-export function getReferenceString(resource: Resource): string {
-  return resource.resourceType + '/' + resource.id;
+export function getReferenceString(resource: Reference | Resource): string {
+  const { reference, resourceType, id } = resource as any;
+  if (reference && typeof reference === 'string') {
```

Comment:
I think you can use the `isReference` [typeguard](https://github.com/medplum/medplum/blob/97a11229f78dba75116758af36813212f95b1100/packages/core/src/types.ts#L334-L337) here